accepted/2024/improve-support-for-packages-signed-with-trusted-signing.md

```diff
@@ -0,0 +1,155 @@
+of these changes.
+
+### Step 2: Change how Trusted Signing EKU's are shown in the NuGet Gallery.
+
+#### How would this be implemented?
+
+The next steps would be to change how this will be displayed inside the NuGet Gallery. Below is an image of how this could be
+added to the account page of a user (please forgive me for my css skills). The naming is a proposal and I am open to suggestions
+but I think it would be best to show the EKU's in a separate table.
+
+![User profile](images/trusted-signing-profile.png)
+
+This image demonstrates how this would look when a user has both an EKU and certificate linked to their account. As mentioned in
+step 1 the certificate would no longer be shown and the user would only see their EKU's. The Subject and Issuer will be extracted
+from the certificate the first time and added to the EKU data. Future uploads should also check the Subject and Issuer and update
+the data if they have changed.
+
+#### How does this help the user?
+
+When only step 1 is implemented it will already be a huge help for users that use Trusted Signing but it will look strange inside
+the NuGet Gallery because no certificates are shown anymore and they cannot unlink their EKU. Adding this separate table with EKU's
+linked to the user their account would allow them to remove this EKU. I personally don't think users would really need this
+functionality right away so that is why I think this could be added as a second step.
+
+### Step 3: Allow a user to submit a Public Trust identity EKU in the NuGet Gallery
+
+#### How would this be implemented?
+
+When a `.cer` file is uploaded only the thumbprint is stored because the certificate is not parsed by the NuGet Gallery due to
+security reasons. This means we cannot add the Public trust identity EKU record when a certificate is uploaded. For a user that
+has access to the Azure Portal it would be easier to copy the EKU from the interface:
+
+![Azure Portal](images/trusted-signing-azure.png)
+
+And this can also be done before a NuGet package has been signed. This copied value should then be entered into a form that adds
+this EKU to the user. As mentioned earlier the Subject and Issuer will be set once the first package that has this EKU is uploaded.
+
+#### How does this help the user?
+
+Adding this in the interface would make it possible for a user to register an EKU in their account. This would be easier than
+extracting the certificate from a signed NuGet package. The process to get that `.cer` file from a signed package is not that
+simple. This would be the final step that would give a user full control over their EKU's.
+
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
```

Comment:
Isn't an alternative solution that we integrate with Trusted Signing and retrieve the subscriber ID directly?

Reply:
That would still require some kind of integration with the Azure API that would require authorization setup. That feels more complicated?

Reply:
I'm not suggesting this as a replacement for the current proposal.

The purpose of this section is to present alternatives and rationale "[w]hy this is the best design compared to other designs." Complexity and cost are solid motivators.

When someone looks back and wonders if we considered option ___, ideally, it would have been captured in this section.

Reply:
I can also help with testing as I am also onboarded with Trusted Signing.